src/miner.cpp

```diff
@@ -718,18 +721,33 @@ void SplitCoinStakeOutput(CBlock &blocknew, int64_t &nReward, bool &fEnableStake
 
             // Push to an output the (reward times the allocation) to the address, increment the accumulator for allocation,
             // decrement the remaining stake output value, and increment outputs used.
+
             SideStakeScriptPubKey.SetDestination(address.Get());
-            
+
             // It is entirely possible that the coinstake could be from an address that is specified in one of the sidestake entries
             // if the sidestake address(es) are local to the staking wallet. There is no reason to sidestake in that case. The
             // coins should flow down to the coinstake outputs and be returned there. This will also simplify the display logic in
             // the UI, because it makes the sidestake and coinstake outputs disjoint from an address point of view.
             if (SideStakeScriptPubKey == CoinStakeScriptPubKey)
                 continue;
-            blocknew.vtx[1].vout.push_back(CTxOut(nReward * iterSideStake->second, SideStakeScriptPubKey));
+
+            int64_t nSideStake = 0;
+
+            // For allocations ending less than 100% assign using sidestake allocation.
+            if (dSumAllocation + iterSideStake->second < 1.0)
+                nSideStake = nReward * iterSideStake->second;
+            // We need to handle the final sidestake differently in the case it brings the total allocation up to 100%,
+            // because testing showed in corner cases the output return to the staking address could be off by one Halford.
+            else if (dSumAllocation + iterSideStake->second == 1.0)
```

Comment:
Do we need to account for cases when the sum is >1.0 or is that handled elsewhere?

Reply:
That check has always been there — line 716.

Reply:
Ah, there it is, hidden by the diff.